pipes.js

```diff
@@ -53,34 +62,36 @@ Pipes = function () {
     })(action);
   }
 
-  if (! _.has(this.actions, pipelineName))
-    throw new Meteor.Error("invalid-pipelinename","pipelineName "+pipelineName+" is not defined");   
+  if (_.indexOf(self._stages, stageName) === -1)
+    throw new Meteor.Error("invalid-stagename","stageName "+stageName+" is not defined");   
 
-  this.actions[pipelineName].push(action);
+  var cue = self.actions[pipeName] || {};
+  cue[stageName] = cue[stageName] || [];
+  cue[stageName].push(action);
+  self.actions[pipeName] = cue;
 };
 
 /**
- * Executes all actions specified in each pipeline in order.   
+ * Executes all actions specified in a pipe for all defined stages, in order.   
  *
  * @method  Pipes.prototype.do
- * @param {string} pipelines An Array or a String containing the name (or names) of the pipelines to be executed. The pipelines are executed in the order that they appear in the array.
- * @param {object...} options The options for this pipeline sequence, passed as a parameter in the initial action. 
+ * @param {string} pipeName The name of the pipe to be executed. 
+ * @param {object...} options The options for this pipe sequence, passed as a parameter to the initial action. 
  *
- * @returns {object...} result The return result of the last action in the sequence.
+ * @returns {object...} result The return result of the last action in all the includes stage action sequences.
  */
-Pipes.prototype.do = function (pipelines, options) {
+Pipes.prototype.do = function (pipeName, options) {
   var self = this;
 
-  if (_.isString(pipelines))
-    pipelines = [pipelines];
+  if (! _.isString(pipeName))
+    throw new Meteor.Error('do-invalid-pipeName','invalid pipeName, should be a string');
 
-  if (pipelines && !_.isArray(pipelines))
-    throw new Meteor.Error('do-invalid-pipelines','invalid pipelines, should be an array or a string');
+  var _activeStages = _.filter(self._stages, function (stageName) {
+    return self.actions[pipeName] && _.has(self.actions[pipeName], stageName) 
+  });
 
-  _.each(pipelines, function (pipeline) {
-    if (!_.isString(pipeline))
-      throw new Error('invalid pipeline, should be a string');
-    var actions = self.actions[pipeline] || [];
+  _.each(_activeStages, function (stageName) {
+    var actions = self.actions[pipeName][stageName] || [];
```

Comment:
This would be better written:

```javascript
var action = self.actions[pipeName] || {};
_.each(self._stages, function (stage) {
  var actions = action[stageName] || [];
  ...
});
```